### ISSUE TITLE
Name - _ aliasing still broken

### ISSUE THREAD

**Issue by timkpaine** [conda-forge/arrow_odbc-feedstock#45]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Our name aliasing still seems broken, I can reliably get conflicts with `arrow-odbc` that I don't get with `arrow_odbc`
xref: https://github.com/conda-forge/arrow_odbc-feedstock/pull/23

**Comment by pacman82** [comment]
I am sorry for all the suffering I caused. I will think longer in the future about the name to avoid dashes.

**Comment by timkpaine** [comment]
ha no worries, its a shame conda doesn't handle it the same way pypi does, really we just need to figure out the best way to get the pinning to work. Its surprising to me that @dhirschfeld 's change doesn't actually work properly, I can't see anything wrong with it on the surface.

**Comment by dhirschfeld** [comment]
I tested the change in the PR and it definitely worked there.

I have actually seen conflicts on Windows because this package doesn't exist there but `arrow-odbc` is marked as `noarch: python` so `mamba` can find the `noarch` `arrow-odbc` on Windows but then complains that it can't install `arrow_odbc` which is because it doesn't exist on Windows!

Do you think that's what you're seeing also @timkpaine? Are you trying to install on Windows?

**Comment by dhirschfeld** [comment]
> I tested the change in the PR and it definitely worked there.

Obviously not comprehensive, multi-platform testing though :/

The whole not being able to install on Windows even though `mamba` can find `arrow-odbc` was pretty confusing to me and took me a while to figure out, so I think it's probably best to only publish `arrow-odbc` on platforms where `arrow_odbc` exists.

**Comment by timkpaine** [comment]
I'm seeing a similar problem on linux.

**Comment by dhirschfeld** [comment]
> I'm seeing a similar problem on linux, if you depend on arrow-odbc it can't seem to work out the correct arrow_odbc to link it to

Ugh. 😔 
I'll try to figure out what's going on...

**Comment by pacman82** [comment]
What do you mean arrow-odbc does not exist on windows? I do publish wheels to pypi and it does build. Or do u mean 32bit windows?

**Comment by dhirschfeld** [comment]
It wasn't in `conda-forge` but I did actually enable them recently: https://github.com/conda-forge/arrow_odbc-feedstock/pull/42

So, that's not likely to be the cause of failing to solve now.

**Comment by timkpaine** [comment]
I think this line is wrong:
```
        - arrow_odbc ={{ version }}=*_{{ build_num }}
```
If I remember correctly, I get conflicts like when installing on py38 "can't get python 3.9". So that would imply that the * isn't working the way it was intended, such that the solver is getting confused.

**Comment by timkpaine** [comment]
This is still broken so lets drop it, conda really should('ve) solved this at some point but as is it just causes conflicts.

**Comment by dhirschfeld** [comment]
Yeah - I never found the time to investigate so best to drop it if it's broken.
